src/elements/earn/pools/addLiquidity/single/AddLiquiditySingle.tsx

```diff
@@ -87,14 +91,21 @@ export const AddLiquiditySingle = ({ pool }: Props) => {
     }
   };
 
+  useAsyncEffect(async (isMounted) => {
```

Comment:
I'd rather we don't let the observables leak from the API abstraction.

Reply:
I agree we can give the approval an enum so it can choose which contract to take (Something like 3 contracts)
Create an issue after this is merged and I will fix it in the rest as well